code/harness/Tester.cpp

```diff
@@ -504,56 +504,59 @@ pair<milliseconds, milliseconds> Tester::test_fsck_and_user_test(
   // Try mounting the file system so that the kernel can clean up orphan lists
   // and anything else it may need to so that fsck does a better job later if
   // we run it.
-  string mount_opts("");
-  // If ext4, remount with "errors=remount-ro".
-  if (fs_type.compare(kExt4) == 0) {
-    if (!mount_opts.empty()) {
-      mount_opts += ",";
-    }
-    mount_opts += kErrMountRo;
-  }
-  if (mount_device(device_path.c_str(), mount_opts.c_str()) != SUCCESS) {
+  if (mount_device(device_path.c_str(),
+        fs_specific_ops_->GetPostReplayMntOpts().c_str()) != SUCCESS) {
     test_info.fs_test.SetError(FileSystemTestResult::kKernelMount);
   }
   umount_device();
 
-  // TODO(ashmrtn): Change this command for btrfs to use `btrfs check`.
-  string command(TEST_CASE_FSCK + fs_type + " " + device_path + " -- -y 2>&1");
+  // Take the comamnd we are given and redirect stderr to stdout so we can pull
+  // it all out with popen below.
+  string command(fs_specific_ops_->GetFsckCommand(device_path) + " 2>&1");
 
   // Begin fsck timing.
   time_point<steady_clock> fsck_start_time = steady_clock::now();
 
   // Use popen so that we can throw all the output from fsck into the log that
   // we are keeping. This information will go just before the summary of what
   // went wrong in the test.
-  FILE *pipe = popen(command.c_str(), "r");
-  char tmp[128];
-  if (!pipe) {
-    test_info.fs_test.SetError(FileSystemTestResult::kOther);
-    test_info.fs_test.error_description = "error running fsck";
+  if (fs_specific_ops_->AlwaysRunFsck() ||
```

Comment:
Some file systems (like xfs) say you should only run the checker if you think the file system has become corrupted. Is this behavior we should follow?

This denotes whether we should always run the checker or just run it if we had trouble mounting earlier.

Reply:
Yes, we shouldn't be running fsck if mount succeeds.